scripts/typescript_checks.py

```diff
@@ -57,13 +811,123 @@ def validate_compiled_js_dir() -> None:
+    # Remove the empty lines and error explanation lines.
+    prefixes = ('core', 'extensions', 'typings')
+    errors = [x for x in errors if x.startswith(prefixes)]
+    # Remove error explanation lines.
+    errors = [x.split('(', 1)[0] for x in errors]
+    # Remove the duplicate occurrences of the file names.
+    files_with_errors = sorted(set(errors))
+
+    # List of missing files that are neither strictly typed nor present in
+    # TS_STRICT_EXCLUDE_PATHS.
+    files_not_type_strict = []
+    for filename in files_with_errors:
+        if filename not in TS_STRICT_EXCLUDE_PATHS:
+            files_not_type_strict.append(filename)
+
+    # Add "typings" folder to get global imports while compiling.
+    files_not_type_strict.append('typings')
+
+    try:
+        # Update "include" field of tsconfig-strict.json with files those are
+        # neither strict typed nor present in TS_STRICT_EXCLUDE_PATHS.
+        # Example: List "files_not_type_strict".
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'r') as f:
+            strict_ts_config = yaml.safe_load(f)
+            strict_ts_config['include'] = files_not_type_strict
+
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'w') as f:
+            json.dump(strict_ts_config, f, indent=2, sort_keys=True)
+            f.write('\n')
+
+        # Compile tsconfig-strict.json with files those are neither strictly
+        # typed nor present in TS_STRICT_EXCLUDE_PATHS. All those files
+        # present inside include property.
+        os.environ['PATH'] = '%s/bin:' % common.NODE_PATH + os.environ['PATH']
+        validate_compiled_js_dir()
+
+        if os.path.exists(COMPILED_JS_DIR):
+            shutil.rmtree(COMPILED_JS_DIR)
+
+        cmd = ['./node_modules/typescript/bin/tsc', '--project', config_path]
+        process = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, encoding='utf-8')
+
+        # The value of `process.stdout` should not be None since we passed
+        # the `stdout=subprocess.PIPE` argument to `Popen`.
+        assert process.stdout is not None
+        error_messages = list(iter(process.stdout.readline, ''))
+
+        # Update tsconfig-strict.json and set to its intial "include" state
+        # example "include": ["core", "extensions", "typings"].
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'r') as f:
+            strict_ts_config = yaml.safe_load(f)
+            strict_ts_config['include'] = prefixes
+
+        with utils.open_file(STRICT_TSCONFIG_FILEPATH, 'w') as f:
+            json.dump(strict_ts_config, f, indent=2, sort_keys=True)
+            f.write('\n')
+
+        if error_messages:
+            print('\n' + '\n'.join(error_messages))
+            print(
+                str(len([x for x in error_messages if x.startswith(prefixes)]))
+                + ' Errors found during compilation.\n')
+            sys.exit(1)
+        else:
+            print('Compilation successful!')
+
+        yield
```

Comment:
what is this `yield` doing?

Reply:
Done

Reply:
Yield represent upto this part we have done setup part without any execution failure, now whatever yield return it will execute (in this we just pass that). If execution fails execute finally part directly.

can't we use try expect directly, as we are not yielding anything?

Reply:
I'm having trouble parsing your comment, but `yield` is only used for generators, and this function isn't a generator. See https://docs.python.org/3/reference/simple_stmts.html#the-yield-statement

Reply:
Removed, can you please take a look now.

Reply:
No need now, make separate strict ts-config, see [this](https://github.com/oppia/oppia/pull/15578#discussion_r900016964) discussion.